apps/dcos_dns/src/dcos_dns_handler.erl

```diff
@@ -177,90 +177,106 @@ resolve_loop(Begin, Workers, Fun, FailMsg, Zone) ->
 -spec(start_worker(protocol(), upstream(), binary()) -> pid()).
 start_worker(Protocol, Upstream, Request) ->
     Pid = self(),
+    Begin = erlang:monotonic_time(),
+    BeginMilliseconds = erlang:monotonic_time(millisecond),
     proc_lib:spawn(fun () ->
```

Comment:
This lambda function has started being too big.

Reply:
Do you think the fix is to extract it as a named function?

Reply:
Yup, it's just a bit easier to read. Also all anonymous functions (since OTP 22, anonymous functions that capture the environment) are created in runtime.